.githooks/pre-commit

```diff
@@ -0,0 +1,76 @@
+#!/bin/sh
+# default pre-commit hook plus astyle
+
+#
+# An example hook script to verify what is about to be committed.
+# Called by "git commit" with no arguments.  The hook should
+# exit with non-zero status after issuing an appropriate message if
+# it wants to stop the commit.
+
+if git rev-parse --verify HEAD >/dev/null 2>&1
+then
+	against=HEAD
+else
+	# Initial commit: diff against an empty tree object
+	against=$(git hash-object -t tree /dev/null)
+fi
+
+# If you want to allow non-ASCII filenames set this variable to true.
+allownonascii=$(git config --type=bool hooks.allownonascii)
+
+# Redirect output to stderr.
+exec 1>&2
+
+# Cross platform projects tend to avoid non-ASCII filenames; prevent
+# them from being added to the repository. We exploit the fact that the
+# printable range starts at the space character and ends with tilde.
+if [ "$allownonascii" != "true" ] &&
+	# Note that the use of brackets around a tr range is ok here, (it's
+	# even required, for portability to Solaris 10's /usr/bin/tr), since
+	# the square bracket bytes happen to fall in the designated range.
+	test $(git diff --cached --name-only --diff-filter=A -z $against |
+	  LC_ALL=C tr -d '[ -~]\0' | wc -c) != 0
+then
+	cat <<\EOF
+Error: Attempt to add a non-ASCII file name.
+
+This can cause problems if you want to work with people on other platforms.
+
+To be portable it is advisable to rename the file.
+
+If you know what you are doing you can disable this check using:
+
+  git config hooks.allownonascii true
+EOF
+	exit 1
+fi
+
+ASTYLE=astyle
+ASTYLEOPTIONS="../../astyle.options"
+
+case `$ASTYLE --version 2> /dev/null` in
+	Artistic*)
+		;;
+	default)
+		echo "astyle not found. Install it to continue."
+		exit 1
+		;;
+esac
+
+
+## from https://github.com/mellowcandle/astyle_precommit_hook
+files=$(git-diff-index --diff-filter=ACMR --name-only -r --cached $against -- | grep -i '\.c$\|\.cpp$\|\.hpp$\|\.h$')
+for file in $files; do
+    x=`echo $file`
+    if test "x$x" != "x"; then
+		# Remove trailing spaces
+        sed -i 's/[[:space:]]*$//' $file
+
+		$ASTYLE $ASTYLEOPTIONS $file
```

Comment:
nit: there is a double tab here, instead of a single

some code around here also uses spaces :)

Reply:
Oh, i missed that. It's now all spaces.